apis/core/v1beta1/openstackcontrolplane_webhook.go

```diff
@@ -381,6 +382,14 @@ func (r *OpenStackControlPlane) DefaultServices() {
 
 	// Neutron
 	r.Spec.Neutron.Template.Default()
+	// WIP: Error when trying to trigger webhook by modifying the CR.
+	if r.Spec.Neutron.APIOverride.Route == nil {
+		r.Spec.Neutron.APIOverride.Route = &route.OverrideSpec{}
+		openstackcontrolplanelog.Info("Neutron Route Override initialized")
+
+	}
+	r.Spec.Neutron.APIOverride.Route.AddAnnotation(r.Spec.Neutron.Template.GetDefaultRouteAnnotations())
```

Comment:
I think this is the important part and you don't need a new field in neutron struct at all.

Reply:
That is what I was told at first. However, I added it because before that I got "runtime error: invalid memory address or nil pointer dereference " since the r.Spec.Neutron.APIOverride.Route value there is actually null (therefore I cannot use the AddAnnotation function)